programs/clearing_house/src/controller/orders.rs

```diff
@@ -1182,6 +1190,13 @@ pub fn fulfill_order_with_match(
             filler.is_some(),
         )?;
 
+    let mut taker_unsettled_pnl = update_position_and_market_with_fee(
```

Comment:
This ends up calling `update_position` for the amm's position, which means fills between two users will affect the amm's cost basis, which doesn't makes any sense. The only fills that should be affecting the amm position should be in `fulfill_order_with_amm`

Reply:
resolved.